Aagam_Shah/postgreAssignment/src/controller/school.controller.ts

```diff
@@ -0,0 +1,359 @@
+        `select * 
+        from school.student as st, school.subject as sb
+        where st.classid = sb.classid and sb.subjectid = $1`, [id]
+      );
+      if(response.rows.length){
+        ctx.response.status = 200;
+        ctx.body = response.rows;
+        return;
+      }
+      ctx.response.status = 400;
+      ctx.body = "Required Id is not Present";
+      return;
+    }
+    catch(error){
+      ctx.response.status = 500;
+      ctx.body = error.message;
+      return;
+    }
+  }
+  else{
+    ctx.body = "invalid id passed";
+    ctx.response.status = 400;
+    return;
+  }
+};
+
+exports.marksOfStudent = async (ctx: Context) => {
+  var id = ctx.params.id;
+  if(isValidId(id)){
+    try{
+      const response = await db.query(
+        `select * from school.marks where studentid = $1`, [id]
+      );
+      if(response.rows.length){
+        ctx.response.status = 200;
+        ctx.body = response.rows;
+        return;
+      }
+      ctx.response.status = 400;
+      ctx.body = "Required Id is not Present";
+      return;
+    }
+    catch(error){
+      ctx.response.status = 500;
+      ctx.body = error.message;
+      return;
+    }
+  }
+  else{
+    ctx.body = "invalid id passed";
+    ctx.response.status = 400;
+    return;
+  }
+};
+
+exports.highestMarkOfSubject = async (ctx: Context) => {
+  var id = ctx.params.id;
+  if(isValidId(id)){
+    try{
+      const response = await db.query(
+        `select * from school.marks 
+        where subjectid = $1
+        order by marks desc
+        limit 1`, [id]
+      );
+      if(response.rows.length){
+        ctx.response.status = 200;
+        ctx.body = response.rows;
+        return;
+      }
+      ctx.response.status = 400;
+      ctx.body = "Required Id is not Present";
+      return;
+    }
+    catch(error){
+      ctx.response.status = 500;
+      ctx.body = error.message;
+      return;
+    }
+  }
+  else{
+    ctx.body = "invalid id passed";
+    ctx.response.status = 400;
+    return;
+  }
+};
+
+exports.topStudents = async (ctx: Context) => {
+  var number = parseInt(ctx.params.number);
+  if(typeof number === "number"){
+    try{
+      const response = await db.query(
+        `select studentid, sum(marks) as total 
+        from school.marks 
+        group by studentid 
+        order by total desc 
+        limit $1`, [number]
+      );
+      if(response.rows.length){
+        ctx.response.status = 200;
+        ctx.body = response.rows;
+        return;
+      }
+      ctx.response.status = 400;
+      ctx.body = "Required Id is not Present";
+      return;
+    }
+    catch(error){
+      ctx.response.status = 500;
+      ctx.body = error.message;
+      return;
+    }
+  }
+  else{
+    ctx.body = "invalid number passed";
+    ctx.response.status = 400;
+    return;
+  }
+};
+
+exports.createStudent = async (ctx: Context) => {
+  var name = ctx.request.body.name.trim();
+  var classid = ctx.request.body.classid.trim();
+  if(!name || !classid){
```

Comment:
should also take few more properties like name,Dob. can also add types for this.

Reply:
we can use https://joi.dev/api/?v=17.4.0 to validate request body.

Reply:
used